pkg/apis/migration/v1alpha1/condition.go

```diff
@@ -203,6 +204,33 @@ func (r *Conditions) SetCondition(condition Condition) {
 	}
 }
 
+// Set (add/update) the specified condition to the collection.
+func (r *Conditions) SetAndLogCondition(logger logging.Logger, condition Condition, namespace, name string) {
```

Comment:
I'm curious how this interacts with the condition systems. In at least some of our controllers, we _stage_ conditions on every reconcile. 

I'd like to know if this new logging mechanism results in a flurry of log messages on every reconcile, even if no condition changes were really made.